docs/src/pages/api-reference/react.mdx

```diff
@@ -142,6 +146,7 @@ export const OurUploadButton = () => (
 | onUploadBegin          | function                                             | No                      | Added in `v5.4` | callback function for upload begin                                                                                          |
 | config                 | Config                                               | No                      | Added in `v5.7` | object to pass additional configuration for button                                                                          |
 | disabled               | boolean                                              | No                      | Added in `v6.7` | Disables the button.                                                                                                        |
+| onSelect               | `(files: File[]) => void`                            | No                      | Added in `v6.8` | Callback function that runs when files are selected.                                                                        |
```

Comment:
`files` here will include **all** selected files, not just the "newly selected" ones, right? Worth a callout?

Like if I already have 1 file selected and then selects 2 more, the event will include all 3 files and not jsut the 2 I selected this time

Reply:
i think "newly selected files" is the same as "all files"? Since `setFiles()` overwrites instead of merging? 

![image](https://github.com/user-attachments/assets/219bd70c-aac9-4c4f-a260-0daa983b12ca)

Reply:
nevermind, I see what you mean.

Reply:
oh, right... it's only if you paste files that it merges??

Reply:
oh right... I should probably add this in the paste handler then too